Tests/Rules/UseDeclaredVarsMoreThanAssignments.tests.ps1

```diff
@@ -73,5 +73,10 @@ function MyFunc2() {
         It "returns no violations" {
             $noViolations.Count | Should -Be 0
         }
+        
+        It "Does not flag += operator" {
+            $results = Invoke-ScriptAnalyzer -ScriptDefinition '$array=@(); $list | ForEach-Object { $array += $c }' | Where-Object { $_.RuleName -eq $violationName }
```

Comment:
what about the case where `$a` is not initialized? Does that fall into this rule or a different one?

Reply:
Hmm, tricky one. I was briefly thinking about it but threw the though away at first because not classifying `+=` as an assignment was much more important. But you have a good point and I think I could take this into account by classifying `+=` as an assignment if and only if it is not already in the list of assigned variables.